test/commands/app/geturl.test.js

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const TheCommand = require('../../../src/commands/app/get-url')
+const BaseCommand = require('../../../src/BaseCommand')
+
+jest.mock('@adobe/aio-app-scripts')
+const mockScripts = require('@adobe/aio-app-scripts')()
+
+beforeEach(() => {
+  jest.restoreAllMocks()
+})
+
+test('exports', async () => {
+  expect(typeof TheCommand).toEqual('function')
+  expect(TheCommand.prototype instanceof BaseCommand).toBeTruthy()
+})
+
+test('description', async () => {
+  expect(TheCommand.description).toBeDefined()
+})
+
+test('aliases', async () => {
+  expect(TheCommand.aliases).toEqual([])
+})
+
+test('flags', async () => {
+  expect(typeof TheCommand.flags.cdn).toBe('object')
+  expect(typeof TheCommand.flags.cdn.description).toBe('string')
+})
+
+describe('run', () => {
+  beforeEach(() => {
+    mockScripts.getUrls = mockScripts.logs
+    mockScripts.getUrls.mockReset()
+  })
+
+  test('get all action urls', async () => {
+    const command = new TheCommand([])
+    command.error = jest.fn()
+    command.log = jest.fn()
+    const retVal = {
+      runtime: {
+        action: 'https://fake_ns.adobeioruntime.net/api/v1/web/sample-app-1.0.0/action'
+      }
+    }
+    mockScripts.getUrls.mockResolvedValue(retVal)
+    const urls = await command.run()
+    expect(command.error).toHaveBeenCalledTimes(0)
+    expect(mockScripts.getUrls).toBeCalledWith({})
+    expect(urls).toBe(retVal)
```

Comment:
I think in all test cases we want to also test that the url is logged, as this is the intent of the command.
something like
`expect(command.log).toHaveBeenCalledWith(expect.stringContaining(url))`

Reply:
done